File: Proyecto/KotlinAPI/src/main/kotlin/com/puj/admincenter/dto/users/UserDto.kt
package com.puj.admincenter.dto.users

import com.puj.admincenter.domain.users.User

data class UserDto(
    val id: Int,
    val email: String,
    var name: String,
    var username: String,
    var password: String
) {
    companion object {
        fun convert(user: User): UserDto {
            val dto = UserDto(
                id = user.id,
                email = user.email,
                name = user.name,
                username = user.username,
                password = user.password
            )
            return dto
        }
    }
}

File: Proyecto/WebDev/WebDev.Services/Entities/UserDto.cs
namespace WebDev.Services.Entities
{
  public class UserDto
  {
    public int id { get; set; }
    public string email { get; set; }
    public string name { get; set; }
    public string username { get; set; }
    public string password { get; set; }

    private UserDto()
    {

    }

    public static UserDto Build(int id, string email, string name, string username, string password)
    {
      return new UserDto
      {
        id = id,
        email = email,
        name = name,
        username = username,
        password = password
      };
    }
  }
}